docs/code-review.md

```diff
@@ -0,0 +1,330 @@
+            t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+            testDir := t.TempDir()
+            err := functionName(testDir)
+            if tt.expectedErr {
+                assert.Error(t, err)
+            } else {
+                assert.NoError(t, err)
+            }
+        })
+    }
+}
+```
+
+If the test has many use cases, it will be grouped in several tests per result, for example:
+
+```golang
+func Test_FunctionNameThenErr(t *testing.T) {
+    var tests = []struct {
+        name         string
+        resource     Resource
+        expectedErr  bool
+    }{
+        {
+            name:        "when resource is nil then error",
+            resource:    nil,
+        },
+        {
+            name:        "when resource is not valid then error",
+            resource:    MalformedResource{},
+        },
+        {
+            name:        "when resource is not valid because of X",
+            resource:    MalformedResource{},
+        },
+        {
+            name:        "when resource fails because of proxy",
+            resource:    MalformedResource{},
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+            testDir := t.TempDir()
+            err := functionName(testDir)
+            if tt.expectedErr {
+                assert.Error(t, err)
+            } else {
+                assert.NoError(t, err)
+            }
+        })
+    }
+}
+func Test_FunctionNameThenNoErr(t *testing.T) {
+    var tests = []struct {
+        name         string
+        resource     Resource
+    }{
+        {
+            name:        "when resource is type 1",
+            resource:    CorrectResource1{},
+        },
+        {
+            name:        "when resource is type 1 and has x feature",
+            resource:    CorrectResource1{
+                FeatureX: true,
+            },
+        },
+        {
+            name:        "when resource is type 2",
+            resource:    CorrectResource2{},
+        },
+    }
+
+    for _, tt := range tests {
+        t.Run(tt.name, func(t *testing.T) {
+            t.Setenv("ENV_REQUIRED_FOR_TEST", "testValue")
+            testDir := t.TempDir()
+            err := functionName(testDir)
+            if tt.expectedErr {
+                assert.Error(t, err)
+            } else {
+                assert.NoError(t, err)
+            }
+        })
+    }
+}
+```
+
+Each test should create its own scenario, eliminating and leaving the state of the machine in the same state as it was at the start of the test.
+
+#### E2E tests
+
+Each command should have its own set of end to end tests to prove that the main functionality works correctly. An e2e test should be added to a new feature if it adds a new use case or breaks an existing use case.
+**In order to speed up the CI process, all tests are executed in parallel including any new tests**
+
+### Logs
```

Comment:
We might want to mention how they should be used. How an information message is displayed to the user or how an info message is just stored for okteto doctor purposes